### PR TITLE
Update and Delete bulk actions support

### PR DESCRIPTION
**Pull request by jo-asakura**
Introduces support of bulk update and delete actions.

**Update action** expects API to meet the following requirements:
 - URL pattern is `http://yourURL/model`
 - HTTP verb is `PUT`
 - Response is an array of updated records

**Delete action** expects API to meet the following requirements:
 - URL pattern is `http://yourURL/model?ids=[1,2,3,...]`
 - HTTP verb is `DELETE`

### CODE REVIEW

File: src/actions/bulkUpdateAction.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      if (options.waitFor) {
+        processUpdate(newModelData, options, dataManager);
+      }
+      return newModelData;
+    });
+}
+
+export default function bulkUpdateAction (dataManager, options) {
+  return (modelData) => {
+    if (!options.waitFor || !options.performRequest) {
```
Comment (mjswensen) [inline src/actions/bulkUpdateAction.js]: Looks like if `options.waitFor` and `options.performRequest` are both `false`, this action is a no-op. Is that correct?

File: tests/actions/bulkUpdateAction.js [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+      waitFor: false,
+    });
+
+    let byId = this.dataManager.getStateForKey('/model').byId;
+    expect(byId[1]).toExist();
+    expect(byId[1].data.something).toEqual('before-1');
+    expect(byId[2]).toExist();
+    expect(byId[2].data.something).toEqual('before-2');
+
+    return action(updatedRecords.map(i => i.id))
```
Comment (mjswensen) [inline tests/actions/bulkUpdateAction.js]: How does the action know how to manipulate the data when all that is passed is an array of ids?

File: src/actions/bulkUpdateAction.js (hunk 1 above)
Comment (jo-asakura) [inline src/actions/bulkUpdateAction.js]: Correct, it's same thing what a singular update does:
image

There shouldn't be any side effects of that.. unless current update doesn't work as expected 😄

File: tests/actions/bulkUpdateAction.js (hunk 2 above)
Comment (jo-asakura) [inline tests/actions/bulkUpdateAction.js]: It doesn't care about your input, it acts as a proxy and API should respond with an array of proper model then `r-d-a` uses `idAttribute` to detect ids.

**Pull request merged**